scipy/optimize/slsqp/slsqp_optmz.f

```diff
@@ -667,65 +670,84 @@ SUBROUTINE lsq(m,meq,n,nl,la,l,g,a,b,xl,xu,x,y,w,jw,mode)
 
       ig = id + meq
 
-      IF (mineq .GT. 0) THEN
-
 C  RECOVER MATRIX G FROM LOWER PART OF A
+C  The matrix G(mineq+2*n,m1) is stored at w(ig)
+C  Not all rows will be filled if some of the upper/lower
+C  bounds are unbounded.
+
+      IF (mineq .GT. 0) THEN
 
           DO 30 i=1,mineq
               CALL dcopy_ (n, a(meq+i,1), la, w(ig-1+i), m1)
    30     CONTINUE
 
       ENDIF
 
-C  AUGMENT MATRIX G BY +I AND -I
-
-      ip = ig + mineq
-      DO 40 i=1,n
-         w(ip-1+i) = ZERO
-         CALL dcopy_ (n, w(ip-1+i), 0, w(ip-1+i), m1)
-   40 CONTINUE
-      w(ip) = one
-      CALL dcopy_ (n, w(ip), 0, w(ip), m1+1)
-
-      im = ip + n
-      DO 50 i=1,n
-         w(im-1+i) = ZERO
-         CALL dcopy_ (n, w(im-1+i), 0, w(im-1+i), m1)
-   50 CONTINUE
-      w(im) = -one
-      CALL dcopy_ (n, w(im), 0, w(im), m1+1)
-
       ih = ig + m1*n
+      iw = ih + mineq + 2*n
 
       IF (mineq .GT. 0) THEN
 
 C  RECOVER H FROM LOWER PART OF B
+C  The vector H(mineq+2*n) is stored at w(ih)
 
           CALL dcopy_ (mineq, b(meq+1), 1, w(ih), 1)
           CALL dscal_sl (mineq,       - one, w(ih), 1)
 
       ENDIF
 
+C  AUGMENT MATRIX G BY +I AND -I, AND,
 C  AUGMENT VECTOR H BY XL AND XU
+C  NaN value indicates no bound
 
+      ip = ig + mineq
       il = ih + mineq
-      CALL dcopy_ (n, xl, 1, w(il), 1)
-      iu = il + n
-      CALL dcopy_ (n, xu, 1, w(iu), 1)
-      CALL dscal_sl (n, - one, w(iu), 1)
+      nancnt = 0
+
+      DO 40 i=1,n
+         if (xl(i).eq.xl(i)) then
+            w(il) = xl(i)
+            do 41 j=1,n
+               w(ip + m1*(j-1)) = 0
+ 41         continue
+            w(ip + m1*(i-1)) = 1
+            ip = ip + 1
+            il = il + 1
+         else
+            nancnt = nancnt + 1
+         end if
+   40 CONTINUE
 
-      iw = iu + n
+      DO 50 i=1,n
+         if (xu(i).eq.xu(i)) then
+            w(il) = -xu(i)
+            do 51 j=1,n
+               w(ip + m1*(j-1)) = 0
+ 51         continue
+            w(ip + m1*(i-1)) = -1
+            ip = ip + 1
+            il = il + 1
+         else
+            nancnt = nancnt + 1
+         end if
+ 50   CONTINUE
 
       CALL lsei (w(ic), w(id), w(ie), w(IF), w(ig), w(ih), MAX(1,meq),
-     .           meq, n, n, m1, m1, n, x, xnorm, w(iw), jw, mode)
+     .           meq, n, n, m1, m1-nancnt, n, x, xnorm, w(iw), jw, mode)
 
       IF (mode .EQ. 1) THEN
 
-c   restore Lagrange multipliers
+c   restore Lagrange multipliers (only for user-defined variables)
```

Comment:
The "non-user-defined" multipliers correspond to the bound constraints
(not to the user-defined inequality/equality constraints). These
Lagrange multipliers are not subsequently used by the routine.
They are set to NAN, so that if they would be used, that would be obvious.